### PR TITLE
fix: Fix yaml quote serialization 325

### LINKED ISSUE #325: Yaml files generated with synth do not quote strings which can be interpreted as native non-string types
### Description of the bug:

When `app.synth()` generates `yaml` files, values which are strings in memory are rendered without quotes in the `yaml` files. This can cause problems for string values such as `"true"`, `"yes"`, and `"2020-01-01"`. These strings, when rendered without quotes, will be read as native data types by `yaml` parsers such as `PyYaml` (`true` and `yes` become python `bool`, while `2020-01-01` becomes `datetime.datetime`).

### Reproduction Steps:

```py
import cdk8s
import yaml

# Create the chart with the unquoted values
app = cdk8s.App()
chart = cdk8s.Chart(app, "test")
cdk8s.ApiObject(chart, "foo", api_version="v1", kind="Foo", metadata={"annotations": {"bar": "2020-01-01", "baz": "on"}})
app.synth()

# Reread the generated file with pyyaml
with open('dist/test.k8s.yaml', 'r') as f:
    objects = list(yaml.safe_load_all(f))

# Print out some info about the re-read types
for key in ["bar", "baz"]:
    print(f"--> {key}")
    print(objects[0]["metadata"]["annotations"][key])
    print(type(objects[0]["metadata"]["annotations"][key]))
```

**Output**
```
--> bar
2020-01-01
<class 'datetime.date'>
--> baz
True
<class 'bool'>
```

**Rendered yaml**
```yaml
apiVersion: v1
kind: Foo
metadata:
  annotations:
    bar: 2020-01-01
    baz: on
  name: test-foo-657aa33b
```

### Error Log:

There are no immediate errors. See the rendered yaml and script output above.

### Environment:

  - **Framework Version**:

```
pip freeze | grep cdk8s
cdk8s==0.27.0
```

  - **OS**: Mac OS, `conda` w/ python 3.6

### Other:

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by gabe-l-hart**
*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Description

This PR addresses https://github.com/awslabs/cdk8s/issues/325 by upgrading the version of the `yaml` package and enabling the option to use double-quotes when serializing strings.

Fixes #325 

## Testing

A unit test has been added and existing tests have been updated to ensure that string values are quoted correctly.

## Known Problems

Upgrading the version of `yaml` has broken the current functionality for serializing `undefined` values. The current test for `undefined` expects that it will be converted to `null`. Version `2.0.0-0` of `yaml` instead drops these values. [On the current `master`](https://github.com/eemeli/yaml/tree/master), there are two ways that this could be fixed: (1) set the [`keepUndefined` option](https://github.com/eemeli/yaml/blob/master/docs/03_options.md#yamldocumentdefaults) or (2) implement [a custom `replacer`](https://github.com/eemeli/yaml/blob/master/docs/02_parse_stringify.md#yamlstringifyvalue-replacer-options---string). 

Neither of these methods is available in `2.0.0-0`, so this PR will likely not be usable until the next release of `yaml` is out with those features enabled.

### CODE REVIEW

File: packages/cdk8s/package.json
Comment (eemeli) [inline packages/cdk8s/package.json:64]:
```suggestion
    "yaml": "2.0.0-0"
```
I would strongly recommend against using a `^` range with prerelease versions. `2.0.0-0` should be usable even in production, but I make no guarantees that the API won't change in `2.0.0-1`.

**Review by eemeli — CHANGES_REQUESTED** [review]

File: packages/cdk8s/package.json
Comment (eladb) [reply packages/cdk8s/package.json:64]: I am okay with that

File: packages/cdk8s/src/yaml.ts
Comment (eladb) [inline packages/cdk8s/src/yaml.ts:28]: So what's the plan?

File: packages/cdk8s/package.json
Comment (gabe-l-hart) [reply packages/cdk8s/package.json:64]: Yes, good point. Thanks for the suggestion

File: packages/cdk8s/src/yaml.ts
Comment (gabe-l-hart) [reply packages/cdk8s/src/yaml.ts:28]: I think for now, we need to hold on this fix until the `yaml` version has been released to include `keepUndefined`. @eemeli It looks like you're a/the primary maintainer for the `yaml` package. Do you have a timeline for when a version will be released with `keepUndefined`?

Comment (eladb) [reply packages/cdk8s/src/yaml.ts:28]: Maybe we can convert all `undefined`s to `null`s?

Comment (eemeli) [reply packages/cdk8s/src/yaml.ts:28]: Soon-ish, esp. as there's clearly a need for it. :) I'm also working on some pretty intensive parser changes, so it'd make sense to wrap up the current state of affairs into a release before those get incorporated. For more specificity, maybe within a week or so?

Comment (gabe-l-hart) [reply packages/cdk8s/src/yaml.ts:28]: @eemeil, I think within a week should work for my team's timing. Thanks!

@eladb I'll update this PR once the `yaml` version is published. I think that's probably better than working around the issue in the short term since this feature has already been implemented in the library.

**Comment by gabe-l-hart** [comment]
@eladb I do see that the [`integration tests` section of the build](https://github.com/awslabs/cdk8s/pull/327/checks?check_run_id=1188349677) is failing. It looks to me like the diff it shows has to do with a failing snapshot test in the `imports` functionality. I don't _think_ the change in this PR should have effected that section of code, but I could imagine that rebuilding the `yarn.lock` could have accidentally bumped the version of a different package and caused this failure. Do you have any more insight on what would have caused this test to fail? Thanks!

**Comment by gabe-l-hart** [comment]
@eladb I was able to figure out the failures. Syncing up with the latest `master` fixed integration tests. I did need to update the integration snapshots to include quotes around the serialized yaml file strings which is expected with this change.

**Review by eladb — APPROVED** [review]

File: packages/cdk8s/src/yaml.ts
Comment (eemeli) [reply packages/cdk8s/src/yaml.ts:28]: [`yaml@2.0.0-1`](https://github.com/eemeli/yaml/releases/tag/v2.0.0-1) is now available.

Comment (eladb) [reply packages/cdk8s/src/yaml.ts:28]: Cool, @gabe-l-hart let me know when ready

Comment (gabe-l-hart) [reply packages/cdk8s/src/yaml.ts:28]: @eemeli Thank you! I'll make the updates and make sure the tests look good

**Comment by gabe-l-hart** [comment]
@eladb I've updated for the latest `yaml` version, enabled the `keepUndefined` flag (with handling for fully `undefined` docs), and updated the tests to match their original checks. I think this will need another ✅  check from you before being ready to merge. Thanks!

**Review by eladb — APPROVED** [review]

**Pull request merged**
